modules/library/main/src/main/java/org/geotools/util/ColorConverterFactory.java

```diff
@@ -42,67 +42,81 @@
+                return target.cast(new Color(rgba, alpha != 0));
+            } else {
+                return null;
+            }
+        }
+    };
+    
+    /**
+     * Converts color to hex representation.
+     */
+    private Converter CONVERT_COLOR_TO_STRING = new Converter() {
+        
+        public <T> T convert(Object source, Class<T> target) throws Exception {
+            Color color = (Color) source;
+            
+            String redCode = Integer.toHexString(color.getRed());
+            String greenCode = Integer.toHexString(color.getGreen());
+            String blueCode = Integer.toHexString(color.getBlue());
+
+            if (redCode.length() == 1) {
+                redCode = "0" + redCode;
+            }
 
-    public Converter createConverter(Class source, Class target, Hints hints) {
+            if (greenCode.length() == 1) {
+                greenCode = "0" + greenCode;
+            }
+
+            if (blueCode.length() == 1) {
+                blueCode = "0" + blueCode;
+            }
+            String hex = ("#" + redCode + greenCode + blueCode).toUpperCase();
+            return target.cast( hex );
+        }
+    };
+
+    public Converter createConverter(Class<?> source, Class<?> target, Hints hints) {
         if (target.equals(Color.class)) {
             // string to color
             if (source.equals(String.class)) {
-                return new Converter() {
-                    public Object convert(Object source, Class target) throws Exception {
-                        String rgba = (String) source;
-                        try {
-                            return Color.decode(rgba);
-                        } catch (NumberFormatException badRGB) {
-                            // unavailable
-                            return null;
-                        }
-                    }
-                };
+                return CONVERT_STRING;
             }
-
             // can we convert the thing to a Integer with a safe conversion?
             if (Number.class.isAssignableFrom(source)) {
-                return new Converter() {
-                    public Object convert(Object source, Class target) throws Exception {
-                        Number number = (Number) source;
-                        // is it an integral number, and small enough to be an integer?
-                        if (((int) number.doubleValue()) == number.doubleValue() && number.doubleValue() < Integer.MAX_VALUE) {
-                            int rgba = number.intValue();
-                            int alpha = 0xff000000 & rgba;
-                            return new Color(rgba, alpha != 0);
-                        } else {
-                            return null;
-                        }
-                    }
-                };
+                return CONVERT_NUMBER;
             }
         } else if (target.equals(String.class) && source.equals(Color.class)) {
```

Comment:
Do we support subclasses of Color? You didn't write this line, but while we're in here, perhaps the Color type equality check should be an assignable check.